### ISSUE TITLE
How to debug this application in PhpStorm

### ISSUE THREAD

**Issue by dzschille** [holodny/docker_codeigniter_lemp#1]
My steps how i could debug the app:

First consideration: in _phpdocker/php-fpm/php-ini-overrides.ini_ i set **xdebug.remote_port** to **9000** which is [xdebugs default](https://xdebug.org/docs/all_settings#remote_port).
An other option would be to change the port in PhpStorms debug configuration.

How to start the app:
```
composer install

docker-compose up -d
```
- Open the app in the browser: http://localhost:8074/
- With [a browser extension](https://github.com/BrianGilbert/xdebug-helper-for-firefox) enable debugging.
- In PhpStorm open the project.
- In PhpStorm click "Start Listening for PHP debug connections".
- Reload the page in the browser.
- Now PhpStorm reacts and tells you that you should set up the server. If there is no reaction, check if your local IP as seen from the webserver container:
```
        ## open console in webserver and view your host IP
        docker-compose exec webserver /bin/sh -c "/sbin/ip route|awk '/default/ { print \$3 }'"
```
Now take this IP and set it in _docker-compose.yml_ as value **XDEBUG_CONFIG**. Then you have to stop your docker containers and start them with `docker-compose up -d` again. You have to use "up" because "start" does not handle the config change.

- Set the server name according to the one you have set as env var in _docker-compose.yml_. Als set the path mapping. The project root to _/application_ is sufficient:
![debugging-1-configure-server](https://user-images.githubusercontent.com/572402/36048664-46a4eb06-0de0-11e8-8162-437006f751a1.png)
- Now you can debug the code:
![debugging-2-stepping](https://user-images.githubusercontent.com/572402/36048764-aae44562-0de0-11e8-8013-d2b61cafe922.png)
- You can also set break points:
![debugging-3-break-point](https://user-images.githubusercontent.com/572402/36048775-b5b5727c-0de0-11e8-8057-5dc5a8252381.png)
- Here a screenshot where you can change the xdebug port, if you want to listen to another one than **9000**:
![debugging-4-phpstorm-debug-config](https://user-images.githubusercontent.com/572402/36048828-d7d2e452-0de0-11e8-894a-33d3407a5d91.png)

**Comment by programetica** [comment]
@dzschille  Thanks!  I feel like I'm almost there.  I went back and started fresh from your instructions without doing any of the Docker integration stuff in IntelliJ.  I followed your instructions and it's not working for me yet -- one thing I notice in my IntelliJ is that when I click on the "Start listening for PHP Debug Connections" button, the square "Stop" button next to it doesn't turn red as it does in your IDE.

I'm going to research that and see what might be going on.  I may be missing a setting or something.

**Comment by dzschille** [comment]
@holodny when I click on the "Start listening for PHP Debug Connections" button, the square "Stop" button next to it doesn't turn red directly. It turns red when the debug session started, so after reloading the web page and the reaction of PhpStorm.
Perhaps it has to do with the IDE? Download a test version of PhpStorm, so you can check if it has to do with IntelliJ or it's settings.

I also updated my HowTo: when you have changed to Xdebug remote host IP in your docker-compose.yml, you have to start the containers with `docker-compose up -d`, "start" is not enough.

You can verify if the setting is active like this:
```
## open console in php container and print env var
docker-compose exec php-fpm bash -c 'echo $XDEBUG_CONFIG'
```

**Comment by programetica** [comment]
@dzschille Ok, I tried it with PHPStorm and had the same results with a new project.  I'm going to try some simpler projects with just PHP & maybe Apache and no configuration changes and see how that works.  I'll update here and at the phpdocker.io repository with my results.  Thanks so much Dzschille!

**Comment by dzschille** [comment]
@holodny , i don't think it has to do with the project code. When no debug session starts it has mostly to do with the Xdebug config, and the IP to connect back. Have you checked your IP in the webserver container?:
`
docker-compose exec webserver /bin/sh -c "/sbin/ip route|awk '/default/ { print \$3 }'"
`
Is it the same you have set in your docker-compose.yml?

**Comment by programetica** [comment]
@dzschille ,  Yah, I did that and added that IP to the yml file as in your example.  I went through your instructions step by step for PHPStorm.  I may be missing something.  I'm looking at where other people have mentioned aliasing the host IP address also.

**Comment by dzschille** [comment]
@holodny if you use docker on mac you can use `docker.for.mac.host.internal` as alias fir your host IP address, [see here](https://docs.docker.com/docker-for-mac/networking/#known-limitations-use-cases-and-workarounds). If you use linux you can not use that. However, you can check your IP like this:
`
docker-compose exec webserver /bin/sh -c "/sbin/ip route|awk '/default/ { print \$3 }'"
`
